core/src/main/java/org/infinispan/interceptors/CacheLoaderInterceptor.java

```diff
@@ -93,46 +87,22 @@ protected void injectDependencies(PersistenceManager clm, EntryFactory entryFact
 
    @Override
    public Object visitApplyDeltaCommand(InvocationContext ctx, ApplyDeltaCommand command) throws Throwable {
-      if (enabled) {
-         Object key;
-         if ((key = command.getKey()) != null) {
-            loadIfNeeded(ctx, key, command);
-         }
-      }
-      return invokeNextInterceptor(ctx, command);
+      return visitDataCommand(ctx, command);
    }
 
    @Override
    public Object visitPutKeyValueCommand(InvocationContext ctx, PutKeyValueCommand command) throws Throwable {
-      if (enabled) {
-         Object key;
-         if ((key = command.getKey()) != null) {
-            loadIfNeeded(ctx, key, command);
```

Comment:
IMO we should have this:
* the put should increment if miss because we return the previous value
* the InvalidateCommand (and subclasses) should not increment (internal use on L1 and state transfer)
* the PutMapCommand should not increment either (we don't need the previous values. actually, it does not need to interact with the cache loader)
* the delta is more complicated but I think it can count the misses. in the end, we need to load the object to apply the deltas.